src/NuGet.Clients/VsConsole/Console/ScriptExecutor.cs

```diff
@@ -175,7 +175,18 @@ public bool TryMarkVisited(PackageIdentity packageIdentity, PackageInitPS1State
             // to switch to powershell pipeline execution thread. In order not to block the UI thread,
             // go off the UI thread. This is important, since, switches to UI thread,
             // using SwitchToMainThreadAsync will deadlock otherwise
-            await Task.Run(() => host.Execute(console, request.BuildCommand(), request.BuildInput()));
+            await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                // we don't yield for a while, so in case we're in an inlined continuation,
+                // give our caller a chance to execute other continuations to avoid deadlocks.
+                await Task.Yield();
+
+                // We need the powershell script to run on the UI thread.
```

Comment:
You might elaborate on this code comment for future engineers who are considering changing this by adding "because the script often interacts with DTE or other VS services, which are STA objects and require the UI thread."